### PR TITLE
[Docs] Mention FSDP in `HuggingFaceTrainer` docs

### PR DESCRIPTION
**Pull request by Yard1**
Signed-off-by: Antoni Baum <antoni.baum@protonmail.com>

## Why are these changes needed?

Mentions that it is possible to run FSDP using `HuggingFaceTrainer`.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/train/huggingface/huggingface_trainer.py [hunk 1]
```diff
@@ -120,7 +120,11 @@ class HuggingFaceTrainer(TorchTrainer):
     This Trainer runs the ``transformers.Trainer.train()`` method on multiple
     Ray Actors. The training is carried out in a distributed fashion through PyTorch
     DDP. These actors already have the necessary torch process group already
-    configured for distributed PyTorch training.
+    configured for distributed PyTorch training. If you have PyTorch >= 1.12.0
+    installed, you can also run FSDP training by specifying the ``fsdp`` argument
+    in ``TrainingArguments``. For more information on configuring FSDP,
+    refer to `Hugging Face documentation <https://huggingface.co/docs/transformers/\
+main/en/main_classes/trainer#transformers.TrainingArguments.fsdp>`__.
```
Comment (amogkam) [inline python/ray/train/huggingface/huggingface_trainer.py:127]: This link is broken for me?

**Review by amogkam — APPROVED** [review]

**Comment by Yard1** [comment]
@amogkam I think it's due to the line break (`\`). I just checked, it works fine without it, make sure you remove it when pasting. It also works fine when you click on it in the docs.
`https://huggingface.co/docs/transformers/main/en/main_classes/trainer#transformers.TrainingArguments.fsdp`

**Comment by amogkam** [comment]
Oh of course 🤦 

Just a nit, but would it make sense to put the full URL on one line with a noqa? I can imaging other users trying to copy-paste directly without realizing there is a line break.

File: python/ray/train/huggingface/huggingface_trainer.py (hunk 1 above)
Comment (amogkam) [inline python/ray/train/huggingface/huggingface_trainer.py:127]: Also seems like the .fsdp doesn't work- it just redirects me to the main Trainer doc

```suggestion
main/en/main_classes/trainer#transformers.TrainingArguments>`__.
```

**Comment by Yard1** [comment]
The noqa would have to apply to the entire docstring, as we can't have a comment inside of it.

**Pull request merged**